### PR TITLE
feat(cb2-8667): merge trl changes into VTMDEV-1

### PR DESCRIPTION
**Pull request by naathanbrown**
## Ticket title

Merge into VTMDEV-1 stage one of the changes to VTM

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts [hunk 1]
```diff
@@ -89,6 +85,14 @@ export class AmendVrmComponent implements OnDestroy, OnInit {
...
   }
 
+  // get makeAndModel(): string | undefined {
+  //   return this.techRecord ? this.technicalRecordService.getMakeAndModel(this.techRecord) : undefined;
+  // }
+
+  // get currentVrm(): string | undefined {
+  //   return this.techRecord?.techRecord_vehicleType !== 'trl' ? this.techRecord?.primaryVrm ?? '' : undefined;
+  // }
+
```
Comment (tomcrawleyy) [inline src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts:95]: possible could be removed if not necessary anymore

File: src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts [hunk 2]
```diff
@@ -34,20 +32,6 @@ export class EditTechRecordButtonComponent implements OnDestroy {
     private routerService: RouterService
   ) {}
 
-  // ngOnInit() {
-  //TODO: V3 is this needed? causing issues when updating record feels like the logic should be handled in the parent
```
Comment (kcarr-bjss) [inline src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts:38]: Can we get rid of this commented out code or does it need to be uncommented and fixed?

File: src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts [hunk 3]
```diff
@@ -72,31 +73,19 @@ describe('TechRecordChangeVrmComponent', () => {
...
   });
 
-  // describe('makeAndModel', () => {
-  //   beforeEach(() => {
-  //     expectedTechRecord = {systemNumber: 'foo', createdTimestamp: 'bar', vin: 'testVin'};
-  //     component.techRecord = expectedTechRecord;
-  //   });
-  //   // TODO: V3 PSV HGV TRL
-  //   // it('should should return the make and model', () => {
-  //   //   expect(component.makeAndModel).toBe(`${expectedTechRecord.chassisMake} - ${expectedTechRecord.chassisModel}`);
```
Comment (kcarr-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts:82]: Can we remove these commented out tests or do they need fixing?

File: src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.ts [hunk 4]
```diff
@@ -69,6 +66,14 @@ export class AmendVinComponent implements OnDestroy, OnInit {
     return this.techRecord ? this.technicalRecordService.getVehicleTypeWithSmallTrl(this.techRecord) : undefined;
   }
 
+  get makeAndModel(): string | undefined {
```
Comment (kcarr-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.ts:69]: Is this actually used anywhere anymore?

File: src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.spec.ts [hunk 5]
```diff
@@ -71,27 +72,6 @@ describe('TechRecordChangeVrmComponent', () => {
...
     expect(component).toBeTruthy();
   });
-  // TODO V3 PSV
-  // describe('makeAndModel', () => {
-  //   beforeEach(() => {
-  //     expectedVehicle = mockVehicleTechnicalRecord(VehicleTypes.PSV);
-  //     component.vehicle = expectedVehicle;
-  //   });
-
-  //   it('should should return the make and model', () => {
```
Comment (kcarr-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.spec.ts:81]: Same here, do we need to fix or remove these unit tests?

File: src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts (hunk 3 above)
Comment (gjulien-bjss) [reply src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts:82]: I mean this is removing them right?

File: src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts [hunk 6]
```diff
@@ -13,24 +14,23 @@ import { Observable, Subject, map } from 'rxjs';
...
+      this.store.dispatch(getBySystemNumber({ systemNumber: (this.currentTechRecord as TechRecordType<'get'>)?.systemNumber }));
     }
   }
 
   get techRecordHistory$() {
     return this.store.select(selectTechRecordHistory);
   }
 
-  get techRecords$() {
+  //TODO: update the type of TechRecordSearch in cvs-type-definitions to include the new fields on GSI in table
```
Comment (naathanbrown) [inline src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts:32]: Think we have done this?

File: src/app/resolvers/contingency-test/contingency-test.resolver.ts [hunk 7]
```diff
@@ -24,38 +26,49 @@ export class ContingencyTestResolver implements Resolve<boolean> {
     private userService: UserService
   ) {}
 
+  //TODO: remove the anys
```
Comment (tomcrawleyy) [inline src/app/resolvers/contingency-test/contingency-test.resolver.ts:29]: comment no longer required

File: src/app/store/technical-records/reducers/technical-record-service.reducer.ts [hunk 8]
```diff
@@ -29,31 +34,34 @@ import {
...
   updateTechRecordFailure,
-  updateTechRecordSuccess,
-  addSectionState,
-  removeSectionState,
-  clearAllSectionStates,
-  getTechRecordV3Success
+  updateTechRecordSuccess
 } from '../actions/technical-record-service.actions';
+import { vehicleBatchCreateReducer } from './batch-create.reducer';
 //TODO: V3 re-import vehicleBatchCreateReducer from batch-create.reducer
```
Comment (tomcrawleyy) [inline src/app/store/technical-records/reducers/technical-record-service.reducer.ts:52]: comment no longer relevant

File: src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts (hunk 6 above)
Comment (gjulien-bjss) [reply src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts:32]: No

File: src/app/forms/custom-sections/trl-brakes/trl-brakes.component.ts [hunk 9]
```diff
@@ -32,8 +32,8 @@ export class TrlBrakesComponent implements OnInit, OnChanges, OnDestroy {
     this.form = this.dfs.createForm(TrlBrakesTemplate, this.vehicleTechRecord) as CustomFormGroup;
 
     this.form.cleanValueChanges.pipe(debounceTime(400), takeUntil(this.destroy$)).subscribe((event: any) => {
-      if (event?.axles) {
-        event.axles = (event.axles as Axle[]).filter(axle => !!axle?.axleNumber);
+      if (event?.techRecord_axles) {
+        event.techRecord_axles = (event.techRecord_axles as any).filter((axle: any) => !!axle?.axleNumber);
```
Comment (kcarr-bjss) [inline src/app/forms/custom-sections/trl-brakes/trl-brakes.component.ts:36]: Are we happy with the "any"s here?

Comment (shivangidas) [reply src/app/forms/custom-sections/trl-brakes/trl-brakes.component.ts:36]: it's a later job :)

File: src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts [hunk 10]
```diff
@@ -42,21 +44,27 @@ export class PsvBrakesComponent implements OnInit, OnChanges, OnDestroy {
   ngOnInit(): void {
     this.form = this.dfs.createForm(PsvBrakesTemplate, this.vehicleTechRecord) as CustomFormGroup;
 
+    //TODO: remove the anys
```
[the remark this answers is not in the document]
Comment (shivangidas) [reply src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts:46]: need this one for now

File: src/app/resolvers/contingency-test/contingency-test.resolver.ts (hunk 7 above)
Comment (shivangidas) [reply src/app/resolvers/contingency-test/contingency-test.resolver.ts:29]: this too

File: src/app/forms/custom-sections/plates/plates.component.ts [hunk 11]
```diff
@@ -48,11 +51,11 @@ export class PlatesComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   get hasPlates(): boolean {
-    return (this.techRecord as any).techRecord_plates !== undefined && (this.techRecord as any).techRecord_plates.length > 0;
+    return this.techRecord.techRecord_plates?.length ? true : false;
```
Comment (gjulien-bjss) [inline src/app/forms/custom-sections/plates/plates.component.ts:54]:
```suggestion
    return !!this.techRecord.techRecord_plates?.length;
```

File: src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts [hunk 12]
```diff
@@ -42,21 +43,27 @@ export class PsvBrakesComponent implements OnInit, OnChanges, OnDestroy {
...
       .pipe(
         debounceTime(400),
         takeUntil(this.destroy$),
-        mergeMap((event: any) =>
-          event?.brakes?.brakeCodeOriginal ? this.referenceDataStore.select(selectBrakeByCode(event.brakes.brakeCodeOriginal)) : of(undefined)
-        ),
+        mergeMap((event: any) => {
+          tap(event => {
+            console.log(event);
+          });
```
Comment (gjulien-bjss) [inline src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts:54]:
```suggestion
```

**Comment by gjulien-bjss** [comment]
I have verbal agreement it's ok to merge

**Pull request merged**